rotkehlchen/api/v1/schemas.py

```diff
@@ -1581,11 +1586,29 @@ class ExchangesResourceAddSchema(Schema):
     name = fields.String(required=True)
     location = LocationField(limit_to=SUPPORTED_EXCHANGES, required=True)
     api_key = ApiKeyField(required=True)
-    api_secret = ApiSecretField(required=True)
+    api_secret = ApiSecretField(load_default=b'')
     passphrase = fields.String(load_default=None)
     kraken_account_type = SerializableEnumField(enum_class=KrakenAccountType, load_default=None)
     binance_markets = fields.List(fields.String(), load_default=None)
 
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if (location := data['location']) not in EXCHANGES_WITHOUT_API_SECRET and not data.get('api_secret'):  # noqa: E501
```

Comment:
since you are checking for length here do an actual len(...). Also if it has a default you don't need the get